File: src/test_spark.py
## Imports
from pyspark import SparkConf, SparkContext
 
## Module Constants
APP_NAME = "Test Spark Application"
 
## Closure Functions
 
## Main functionality
 
def main(sc):
	logFile = "test_log"
	rdd = sc.textFile(logFile).cache()

	counts = rdd.map(lambda line: line.split()[8]).map(lambda word: (word, 1)).reduceByKey(test).sortByKey(lambda x: x) 

	output = counts.collect()  
	for (word, count) in output:  
	    print "%s: %i" % (word, count)

	sc.stop()	
def test(a, b):
	print a + "|" + b
	return a + b

if __name__ == "__main__":
	# Configure Spark
	conf = SparkConf().setAppName(APP_NAME)
	conf = conf.setMaster("local[*]")
	sc   = SparkContext(conf=conf)
	# Execute Main functionality
	main(sc)

File: src/test_spark.scala
package demo.spark

import java.io._
import java.lang._

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object TestConnectSpark {
	def main(args: Array[String]): Unit = {
		val sc = new SparkContext("local", "Test Spark Application")
		val rdd = sc.textFile("test_log").cache()
		var counts = rdd.map(x => x.split(" ")(8)).map(x => (x,1)).reduceByKey(a,b => a+b).sortByKey(true)
		var output = counts.collect()  
		for ((word, count) <- output)
			printf("%s: %i", word, count)

		sc.stop()	
	}
}